src/System.Linq/src/System/Linq/Repeat.cs

```diff
@@ -79,10 +79,7 @@ public TResult[] ToArray()
                 TResult[] array = new TResult[_count];
                 if (_current != null)
```

Comment:
`System.Runtime.CompilerServices.RuntimeHelpers.Equals(_current, default(TResult)` is identity-equality for reference types, internal representation quality for value types, ignoring all `Equals` and `==` overrides.